nodestream_akamai/akamai_utils/client.py

```diff
@@ -85,9 +90,13 @@ def _get_api_from_relative_path(
                 response.status_code,
                 response.text,
             )
-        response.raise_for_status()
-        # raise for status only handles: 400 <= status_code < 600
-        raise Exception(f"response.status_code: {response.status_code}", response.text)
+        if response:
+            response.raise_for_status()
+            # raise for status only handles: 400 <= status_code < 600
+            msg = f"Unexpected status {response.status_code} for url: {response.url}"
+            raise HTTPError(msg, response=response)
+        msg = "Missing response object in _get_api_from_relative_path"
+        raise SystemError(msg)
```

Comment:
I _have_ seen this in our logs, where the `response.raise_for_status()` itself throws an error, but it's rare so I'm not sure the real-life circumstances that cause it.